src/libsyntax/tokenstream.rs

```diff
@@ -206,5 +210,1105 @@ impl TokenTree {
                                                          true);
         macro_parser::parse(cx.parse_sess(), cx.cfg(), arg_rdr, mtch)
     }
+
+    /// Check if this TokenTree is equal to the other, regardless of span information
+    pub fn unspanned_eq(&self, other: &TokenTree) -> bool {
+        match (self, other) {
+            (&TokenTree::Token(_, ref tk), &TokenTree::Token(_, ref tk2)) => tk == tk2,
+            (&TokenTree::Delimited(_, ref dl), &TokenTree::Delimited(_, ref dl2)) => {
+                if (*dl).delim == (*dl2).delim {
+                    if dl.tts.len() != dl2.tts.len() {
+                        return false;
+                    }
+                    for (tt1, tt2) in dl.tts.iter().zip(dl2.tts.iter()) {
+                        if !tt1.unspanned_eq(tt2) {
+                            return false;
+                        }
+                    }
+                    true
+                } else {
+                    false
+                }
+            }
+            (_, _) => false,
+        }
+    }
+
+    /// Retrieve the TokenTree's span
+    pub fn span(&self) -> Span {
+        match *self {
+            TokenTree::Token(sp, _) |
+            TokenTree::Delimited(sp, _) |
+            TokenTree::Sequence(sp, _) => sp,
+        }
+    }
+
+    /// Indicates if the token is an identifier
+    pub fn is_ident(&self) -> bool {
+        match *self {
+            TokenTree::Token(_, Token::Ident(_)) => true,
+            _ => false,
+        }
```

Comment:
this should probably be true if the token tree is implicitly delimited around an ident